### PR TITLE
Feature/create account form validation

### LINKED ISSUE #5: Add validations to create account form
# REQUIREMENTS
To create an account valid email and password must be provided on the registration form #4 
Descriptive information in the form of text and change of colors to users so they can easily identify what is wrong with the information they are providing or indicate if everything is ok.

- For email, a valid email must be provided
- For password
  - length of 8 or more characters
  - at least 1 uppercase character
  - at least 1 lowercase character   
  - at least 1 digit character
- Button "Siguiente" should only be enabled when valid information is provided.

A change of color in red will occur when one of the requirements is not met. For instance
-  if a password with a length of fewer than 8 characters is provided. The text "tener 8 o más caracteres" must change to the color red
- if no valid email is provided. The corresponding "required" text must also change to red.

# DESIGN
2 2 Registro (Paso 2)

# Acceptance Criteria
List all required acceptance criteria
- [ ] Form fields validations
- [ ] Submit button is only enabled when valid information is provided
- [ ] Relevant information in the form of change of colors must be presented to users so they can easily notice what is wrong

### PR DESCRIPTION
**Pull request by xErikPsx**
# Description
In this PR were created all the validators for each case of wrong input password and a wrong email, each case gets highlighted in color red if a term is missing and the return to the original color.

#Screenshots

image

closes #5

### CODE REVIEW

**Comment by cfcortes** [comment]
@xErikPsx on PRs Description you must specify or explain the purpose of the intention of your contibution (coder), this is why or what you have implemented and submitted.

I understand that in this case, you submitted your PR so others could have a look at your code and help you. Still should have described your PR, and perhaps you could have added a note or an additional note asking for help t other (mentors).

I think you can still edit your description. I so, please add it.

**Comment by xErikPsx** [comment]
I just tried to upload the PR as soon as I could because Aramando was answering me quickly and I wanted to him checked asap because I know their busy too. But of course, I will edit my PR review.

File: src/screens/RegisterAccount.js [hunk 1]
```diff
@@ -1,43 +1,169 @@
...
+            console.log('>= 8 y minuscula y mayuscula y numero');
+            if (data.password === data.validatePassword) {
+              setColor({...color, passwordInput: false});
+              return true;
+            } else {
+              setColor({...color, passwordInput: true});
+              Alert.alert('Las contraseñas no coinciden');
+              return false;
+            }
+          } else {|
```
Comment (luisarmandodv) [inline src/screens/RegisterAccount.js]:
```suggestion
          } else {
```

File: src/screens/RegisterAccount.js [hunk 2]
```diff
@@ -1,43 +1,169 @@
...
+      setColor({...color, passwordInput: false, firsTerm: false});
+      if (arePasswordsCorrect.lowerCase) {
+        setColor({...color, passwordInput: false, secondTerm: false});
+
+        console.log('>= 8 y minuscula');
+        if (arePasswordsCorrect.caps) {
+          setColor({...color, passwordInput: false, thirdTerm: false});
+          console.log('>= 8 y minuscula y mayuscula');
+          if (arePasswordsCorrect.number) {
+            setColor({...color, passwordInput: false, fourthdTerm: false});
```
Comment (luisarmandodv) [inline src/screens/RegisterAccount.js]:
```suggestion
            setColor({...color, passwordInput: false, fourthTerm: false});
```

File: src/screens/RegisterAccount.js [hunk 3]
```diff
@@ -1,43 +1,169 @@
...
+        }
+      } else {
+        setColor({...color, passwordInput: true, secondTerm: true});
+        console.log('< 8 no minuscula');
+      }
+    } else {
+      setColor({...color, passwordInput: true, firstTerm: true});
+      console.log(' es < 8');
+    }
+  };
```
Comment (luisarmandodv) [inline src/screens/RegisterAccount.js]:
```suggestion
  const validateOnPasswords = () => {
    const { lengthPassword, lowerCase, caps, number } = validatePasswords(
      data.password,
      data.validatePassword,
    );

    const passwordInput = !lengthPassword || !lowerCase || !caps || !number || !(data.password === data.validatePassword);
    
    setColor({
      ...color,
      passwordInput,
      firsTerm: !lengthPassword,
      secondTerm: !lowerCase,
      thirdTerm: !caps,
      fourthTerm: !number
    });

    return passwordInput;
  };
```

**Review by luisarmandodv — CHANGES_REQUESTED** [review]

**Review by OwenRamire — DISMISSED** [review]
🚀

File: src/components/PasswordRequirements.js [hunk 4]
```diff
@@ -34,6 +40,7 @@ const styles = StyleSheet.create({
   },
   textInfo: {
     fontSize: 15,
+    // color: Colors.red
```
Comment (luisarmandodv) [inline src/components/PasswordRequirements.js]:
```suggestion
```

File: src/components/PasswordRequirements.js [hunk 5]
```diff
@@ -1,22 +1,28 @@
 import React from 'react';
 import {View, Text, StyleSheet} from 'react-native';
 import {PASS_REQUIREMENTS} from '../utils/Constants';
+import {Colors} from '../utils/Colors';
 
-const PasswordRequirements = () => {
+const PasswordRequirements = (props) => {
```
Comment (luisarmandodv) [inline src/components/PasswordRequirements.js:7]:
```suggestion
const PasswordRequirements = (props) => {
  const keyItemsArray = ['firstTerm', 'secondTerm', 'thirdTerm', 'fourthTerm'];

  return keyItemsArray.map( key => {
    const color = props[key] ? Colors.red : Colors.black;
    const text = PASS_REQUIREMENTS[key];
    return (
      <Text style={[styles.textInfo, {color}]} key={key}>
        <Text style={styles.dash}>- </Text>
        {text}
      </Text>
      )
    )
  };
```

Comment (luisarmandodv) [reply src/components/PasswordRequirements.js:7]: All that function, can be refactored to this

File: src/screens/RegisterAccount.js [hunk 6]
```diff
@@ -8,38 +8,135 @@ import {
...
+
+  const validateOnInputs = () => {
+    const isEmailEmpty = isEmptyInput(data.email);
+    const isPasswordEmpty = isEmptyInput(data.password);
+    const isValidatePassEmpty = isEmptyInput(data.validatePassword);
+    const isUserPasswordValidation = validateOnPasswords();
+    if (isEmailEmpty && isPasswordEmpty && isValidatePassEmpty) {
+      if (validateOnEmail() === true && isUserPasswordValidation === false) {
+        console.log('Exito');
```
Comment (luisarmandodv) [inline src/screens/RegisterAccount.js]:
```suggestion
```

File: src/screens/RegisterAccount.js [hunk 7]
```diff
@@ -8,38 +8,135 @@ import {
...
+    const isEmailCorrect = validateEmail(data.email);
+    return isEmailCorrect;
+  };
+
+  const validateOnPasswords = () => {
+    const {lengthPassword, lowerCase, caps, number} = validatePasswords(
+      data.password,
+      data.validatePassword,
+    );
+    console.log('validateOnPasswords -> lengthPassword', lengthPassword);
```
Comment (luisarmandodv) [inline src/screens/RegisterAccount.js]:
```suggestion
```

**Review by luisarmandodv — CHANGES_REQUESTED** [review]
Remember to use PropTypes on ALL the components

https://es.reactjs.org/docs/typechecking-with-proptypes.html

File: src/screens/RegisterAccount.js [hunk 8]
```diff
@@ -8,38 +8,135 @@ import {
...
+    const isValidatePassEmpty = isEmptyInput(data.validatePassword);
+    const isUserPasswordValidation = validateOnPasswords();
+    if (isEmailEmpty && isPasswordEmpty && isValidatePassEmpty) {
+      if (validateOnEmail() === true && isUserPasswordValidation === false) {
+        console.log('Exito');
+        navigation.navigate('ConfirmEmail', {
+          usersEmail: data.email,
+        });
+      } else {
+        console.log('Sin exito');
```
Comment (fernandoamz) [inline src/screens/RegisterAccount.js]:
```suggestion
return
```

File: src/screens/RegisterAccount.js [hunk 9]
```diff
@@ -8,38 +8,135 @@ import {
...
     <SafeAreaView style={styles.container}>
       <ScrollView>
         <Text style={styles.title}>Crea tu cuenta</Text>
         <View style={styles.formContainer}>
           <EmailPassTxtInput
+            isCorrect={color.emailInput}
+            id="email"
+            keyboard="email-address"
+            value={data.email}
             title="Correo electrónico"
```
Comment (fernandoamz) [inline src/screens/RegisterAccount.js]: Add to constant file

File: src/screens/RegisterAccount.js [hunk 10]
```diff
@@ -8,38 +8,135 @@ import {
...
             isRequired={IS_REQUIRED}
             password={false}
+            onChangeText={handleOnChange}
           />
           <EmailPassTxtInput
+            isCorrect={color.passwordInput}
+            id="password"
+            keyboard="default"
+            value={data.password}
             title="Clave"
```
Comment (fernandoamz) [inline src/screens/RegisterAccount.js]: Add to constant file

**Review by fernandoamz — CHANGES_REQUESTED** [review]

File: src/components/PasswordRequirements.js [hunk 11]
```diff
@@ -1,22 +1,28 @@
...
+  const firstTermWrong = props.firstTerm ? Colors.red : Colors.black;
+  const secondTermWrong = props.secondTerm ? Colors.red : Colors.black;
+  const thirdTermWrong = props.thirdTerm ? Colors.red : Colors.black;
+  const fourthTermWrong = props.fourthTerm ? Colors.red : Colors.black;
```
Comment (LeyaimMagma) [inline src/components/PasswordRequirements.js]: You can refactor this and create a function

**Review by LeyaimMagma — CHANGES_REQUESTED** [review]

**Review by luisarmandodv — APPROVED** [review]

**Review by LeyaimMagma — APPROVED** [review]

**Review by LeyaimMagma — APPROVED** [review]

**Pull request merged**